docs/user-guide/expressions/lists.md

```diff
@@ -1,47 +1,102 @@
+1. Parse the string input as a `List` of string values (already done).
+2. Identify those strings that can be converted to numbers.
+3. Identify the number of non-numeric values (i.e. `null` values) in the list, by row.
+4. Rename this output as `errors` so that we can easily identify the stations.
 
+The third step requires a casting (or alternately, a regex pattern search) operation to be perform on each element of the list. We can do this using by applying the operation on each element by first referencing them in the `pl.element()` context, and then calling a suitable Polars expression on them. Let's see how:
+
+{{code_block('user-guide/expressions/lists','count_errors',['count_errors'])}}
+```python exec="on" result="text" session="user-guide/lists"
+--8<-- "python/user-guide/expressions/lists.py:count_errors"
+```
+
+What if we chose the regex route (i.e. recognizing the presence of *any* alphabetical character?)
+
+{{code_block('user-guide/expressions/lists','count_errors_regex',['count_errors_regex'])}}
 ```python exec="on" result="text" session="user-guide/lists"
---8<-- "python/user-guide/expressions/lists.py:rank"
+--8<-- "python/user-guide/expressions/lists.py:count_errors_regex"
 ```
 
-## Running polars expression on list elements
+If you're unfamiliar with the `(?i)`, it's a good time to look at the [`str.contains`](https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.Expr.str.contains.html#polars-expr-str-contains) documentation for Polars! The rust regex crate provides a lot of additional regex flags that might come in handy.
+
+# Row-wise computations
+
+This context is ideal for computing in row orientation.
+
+Polars expressions work on columns that have the guarantee that they consist of homogeneous data. Columns have this guarantee, rows in a `DataFrame` not so much. Luckily we have a data type that has the guarantee that the rows are homogeneous: `pl.List` data type.  We can **any** Polars operations on the elements of the list with the `list.eval` (`list().eval` in Rust) expression! These expressions run entirely on polars' query engine and can run in parallel so will be super fast.
 
-We can run **any** polars expression on the elements of a list with the `list.eval` (`list().eval` in Rust) expression! These expressions run entirely on polars' query engine and can run in parallel so will be super fast.
+Let's say we have another set of weather data across three days, for different stations:
+
+{{code_block('user-guide/expressions/lists','weather_by_day',['weather_by_day'])}}
+```python exec="on" result="text" session="user-guide/lists"
+--8<-- "python/user-guide/expressions/lists.py:weather_by_day"
+```
+
+Let's do something interesting. `pandas` allows you to compute the percentages of the rank values. `Polars` doesn't provide such a keyword argument. But because expressions are so versatile we can create our own percentage rank expression for highest temperature. Let's try that!
```

Comment:
Could you remove the comparison with pandas. Users who are not familiar with the package or which are coming from other languages don't understand the reference

Reply:
This is unchanged from the content (save for formatting) in the guide [currently present online;](https://pola-rs.github.io/polars-book/user-guide/expressions/lists/) and it is a good one. Are you sure you want it to go?